File: Dante1/Module1.02/Zadanie2.08/main.c
#include <stdio.h>

int main()
{
    int number;
    char letter;

    printf("Enter smth plz: ");

    int check = scanf("%c%i",&letter,&number);
    if(check != 2)
    {
        printf("Incorrect input");
        return 1;
    }

    if(letter >= 'a' && letter <= 'h' && number >= 1 && number <= 8)
    {
        if((letter == 'a' || letter == 'c' || letter == 'e' || letter == 'g') && (number % 2 == 0))
        {
            printf("White");
        }
        if((letter == 'b' || letter == 'd' || letter == 'f' || letter == 'h') && (number % 2 == 1))
        {
            printf("White");
        }
        if((letter == 'a' || letter == 'c' || letter == 'e' || letter == 'g') && (number % 2 != 0))
        {
            printf("Black");
        }
        if((letter == 'b' || letter == 'd' || letter == 'f' || letter == 'h') && (number % 2 == 0))
        {
            printf("Black");
        }
    }
    else
    {
        printf("Incorrect input");
        return 1;
    }

    return 0;
}

File: Dante1/Module1.02/Zadanie2.08/main.rs
fn main() {
    let mut input: String = String::new();

    println!("Enter: ");

    input.clear();
    std::io::stdin().read_line(&mut input).expect("Incorrect");
    let number: i32 = input.trim().parse().expect("Incorrect");

    input.clear();
    std::io::stdin().read_line(&mut input).expect("Incorrect");
    let letter: char = input.trim().parse().expect("Incorrect");

    if letter >= 'a' && letter <= 'h' && number >= 1 && number <= 8 {
        if (letter == 'a' || letter == 'c' || letter == 'e' || letter == 'g') && (number % 2 == 0) ||
            (letter == 'b' || letter == 'd' || letter == 'f' || letter == 'h') && (number % 2 == 1) {
            print!("White");
        }
        if (letter == 'a' || letter == 'c' || letter == 'e' || letter == 'g') && (number % 2 != 0) ||
            (letter == 'b' || letter == 'd' || letter == 'f' || letter == 'h') && (number % 2 == 0)  {
            print!("Black");
        }
    }
    else {
        panic!("Incorrect");
    }
}
